test/screenshot/chips/index.js

```diff
@@ -2,73 +2,140 @@ import React from 'react';
   }
+}
 
-  handleSelect = (id) => {
-    if (this.isSelected(id)) {
-      this.setState({selectedChipId: -1});
-    } else {
-      this.setState({selectedChipId: id});
-    }
-  }
+class FilterChipsTest extends React.Component {
+  state = {
+    selectedChipIds: this.props.selectedChipIds, // eslint-disable-line react/prop-types
+  };
 
   render() {
+    const {children} = this.props; // eslint-disable-line react/prop-types
     return (
-      <ChipSet>
-        <Chip selected={this.isSelected(0)} id={0} label='Small' handleSelect={this.handleSelect}/>
-        <Chip selected={this.isSelected(1)} id={1} label='Medium' handleSelect={this.handleSelect}/>
-        <Chip selected={this.isSelected(2)} id={2} label='Large' handleSelect={this.handleSelect}/>
-      </ChipSet>
+      <div>
+        <ChipSet
+          filter
+          selectedChipIds={this.state.selectedChipIds}
+          handleSelect={(selectedChipIds) => this.setState({selectedChipIds})}
+        >
+          {children}
+        </ChipSet>
+        <button onClick={() => this.setState({selectedChipIds: ['2chip', '0chip']})}>
+          Select first and last
+        </button>
+      </div>
     );
   }
 }
 
-class ShoppingFilters extends React.Component {
+class InputChipsTest extends React.Component {
   state = {
-    selectedChipIds: new Set([0, 1]),
+    chips: this.props.labels.map((label) => { // eslint-disable-line react/prop-types
+      return {label: label, id: uuidv1()};
+    }),
   };
 
-  isSelected = (id) => {
-    return this.state.selectedChipIds.has(id);
+  addChip(label) {
+    // Create a new chips array to ensure that a re-render occurs.
+    // See: https://reactjs.org/docs/state-and-lifecycle.html#do-not-modify-state-directly
+    const chips = [...this.state.chips];
+    chips.push({label, id: uuidv1()});
+    this.setState({chips});
   }
 
-  handleSelect = (id) => {
-    const selectedChipIds = new Set(this.state.selectedChipIds);
-    if (this.isSelected(id)) {
-      selectedChipIds.delete(id);
-    } else {
-      selectedChipIds.add(id);
+  handleKeyDown = (e) => {
+    if (e.key === 'Enter' && e.target.value) {
+      this.addChip(e.target.value);
+      e.target.value = '';
     }
-    this.setState({selectedChipIds});
   }
 
   render() {
     return (
-      <ChipSet filter>
-        <Chip selected={this.isSelected(0)} id={0} label='Tops' handleSelect={this.handleSelect}/>
-        <Chip selected={this.isSelected(1)} id={1} label='Bottoms' handleSelect={this.handleSelect}/>
-        <Chip selected={this.isSelected(2)} id={2} label='Shoes' handleSelect={this.handleSelect}/>
-      </ChipSet>
+      <div>
+        <input
+          type="text"
+          onKeyDown={this.handleKeyDown}
+        />
+        <ChipSet
+          input
+          // handleRemove removes the chip element from the page
+          handleRemove={(chipId) => {
+            const {chips} = this.state;
+            const chip = chips.find((chip) => chip.id === chipId);
```

Comment:
I think .find is not suppported by IE
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find